core/src/main/scala/org/apache/spark/storage/DiskStore.scala

```diff
@@ -49,7 +49,7 @@ private[spark] class DiskStore(
   private val minMemoryMapBytes = conf.getSizeAsBytes("spark.storage.memoryMapThreshold", "2m")
   private val blockSizes = new ConcurrentHashMap[String, Long]()
 
-  def getSize(blockId: BlockId): Long = blockSizes.get(blockId.name)
+  def getSize(blockId: BlockId): Long = blockSizes.get(blockId)
```

Comment:
Hm, how does this work if the map keys are still strings?

Reply:
Bah, sorry, fixed.